stats.py

```diff
@@ -0,0 +1,45 @@
+from typing import Optional, Dict, Union
+
+StatsAnnot = Dict[str, Union[int, float]]
+
+class Stats:
+    """
+    Champions, items, and rune shards can all be considered to have stats.
+    Stats in league all have a fixed way of being stacked, hence why it makes sense to define a class.
+    """
+
+    def __init__(self, stat_dict: Optional[StatsAnnot] = None):
+        if stat_dict is None:
+            stat_dict = {}
+            
+        for name, value  in stat_dict.items():
+            setattr(self, name, value)
+
+    def print_stats(self):
+        return print("\n".join([f"{k}: {v}" for k, v in self.__dict__.items()]))
+
+
+    def __add__(self, stats: StatsAnnot):
+        addition = dict()
+        for name, value in stats.__dict__.items():
+            if not hasattr(self, name):
+                addition[name] = value
+            else:
+                addition[name] = getattr(self, name) + value
```

Comment:
Where are the multiplicatively stacking stats?

Reply:
here the mult stacking stats would be added normally e.g armor_reduction_perc (15% +25%)

Then the armor reductuon could be handled in a function during an attack to generate flat amount of armor to subtract

Reply:
My point is that you can't add `armor_reduction_perc`. 50% and 50% combine to 75%, not 100%.

Reply:
Oh. I also need to check the behaviour for crit chance attack speed
other wise I guess a if statement on the mult stacking stat name to treat those cases should work

Reply:
I agree that it's probably wiser to make percent armor penetration stack multiplicatively. In New_structure_ideas, the add_stats of Stats has every correct stacking methods

Reply:
👌 I will copy that